src/main/java/io/strimzi/kafka/metrics/DataPointSnapshotBuilder.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.kafka.metrics;
+
+import com.yammer.metrics.core.Histogram;
+import com.yammer.metrics.core.Sampling;
+import com.yammer.metrics.core.Timer;
+import io.prometheus.metrics.model.snapshots.CounterSnapshot;
+import io.prometheus.metrics.model.snapshots.GaugeSnapshot;
+import io.prometheus.metrics.model.snapshots.InfoSnapshot;
+import io.prometheus.metrics.model.snapshots.Labels;
+import io.prometheus.metrics.model.snapshots.PrometheusNaming;
+import io.prometheus.metrics.model.snapshots.Quantile;
+import io.prometheus.metrics.model.snapshots.Quantiles;
+import io.prometheus.metrics.model.snapshots.SummarySnapshot;
+
+import java.util.Arrays;
+
+/**
+ * Builder methods to convert Kafka metrics into Prometheus metrics
+ */
+public class DataPointSnapshotBuilder {
+
+    /**
+     * Convert a non numeric metric into a datapoint for a {@link InfoSnapshot} metric
+     * @param value The value of the datapoint
+     * @param labels Labels associated with the datapoint
+     * @param metricName The name of the new label
+     * @return The {@link InfoSnapshot.InfoDataPointSnapshot} datapoint
+     */
+    public static InfoSnapshot.InfoDataPointSnapshot convert(Object value, Labels labels, String metricName) {
+        labels = labels.add(PrometheusNaming.sanitizeLabelName(metricName), String.valueOf(value));
+        return InfoSnapshot.InfoDataPointSnapshot.builder()
+                .labels(labels)
+                .build();
+    }
+
+    /**
+     * Convert a value into a datapoint for a {@link GaugeSnapshot} metric
+     * @param value The value
+     * @param labels Labels associated with the datapoint
+     * @return The {@link GaugeSnapshot.GaugeDataPointSnapshot} datapoint
+     */
+    public static GaugeSnapshot.GaugeDataPointSnapshot convert(double value, Labels labels) {
+        return GaugeSnapshot.GaugeDataPointSnapshot.builder()
+                .value(value)
+                .labels(labels)
+                .build();
+    }
+
+    /**
+     * Convert a Number into a datapoint for a {@link CounterSnapshot} metric
+     * @param number The value
+     * @param labels Labels associated with the datapoint
+     * @return The {@link CounterSnapshot.CounterDataPointSnapshot} datapoint
+     */
+    public static CounterSnapshot.CounterDataPointSnapshot convert(Number number, Labels labels) {
```

Comment:
I'm not super happy with the various methods especially as `convert(double value, Labels labels)` and `convert(Number number, Labels labels)` shadow each other. Ideally I wanted to keep this file "Yammer free" but I ended up having to use Timer and Histogram.

So maybe we should tweak some of these `convert()` methods to accept the Yammer and Kafka metric types.

Reply:
if you had  sum and count parameters that were Suppliers and a value Function, you could hide yammerness from this class.

Reply:
I shuffled a few things around so this class does not depend on Yammer anymore.